pkg/kubernetes/podtemplatespec/v1alpha1/podtemplatespec.go

```diff
@@ -0,0 +1,272 @@
+// Copyright © 2018-2019 The OpenEBS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	errors "github.com/openebs/maya/pkg/errors/v1alpha1"
+	container "github.com/openebs/maya/pkg/kubernetes/container/v1alpha1"
+	volume "github.com/openebs/maya/pkg/kubernetes/volume/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// PodTemplateSpec holds the api's podtemplatespec objects
+type PodTemplateSpec struct {
+	Object *corev1.PodTemplateSpec
+}
+
+// Builder is the builder object for Pod
+type Builder struct {
+	podtemplatespec *PodTemplateSpec
+	errs            []error
+}
+
+// NewBuilder returns new instance of Builder
+func NewBuilder() *Builder {
+	return &Builder{
+		podtemplatespec: &PodTemplateSpec{
+			Object: &corev1.PodTemplateSpec{},
+		},
+	}
+}
+
+// WithName sets the Name field of podtemplatespec with provided value.
+func (b *Builder) WithName(name string) *Builder {
+	if len(name) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build podtemplatespec object: missing name"),
+		)
+		return b
+	}
+	b.podtemplatespec.Object.Name = name
+	return b
+}
+
+// WithNamespace sets the Namespace field of PodTemplateSpec with provided value.
+func (b *Builder) WithNamespace(namespace string) *Builder {
+	if len(namespace) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New(
+				"failed to build podtemplatespec object: missing namespace",
+			),
+		)
+		return b
+	}
+	b.podtemplatespec.Object.Namespace = namespace
+	return b
+}
+
+// WithAnnotations sets the annotation field of podtemplatespec
+func (b *Builder) WithAnnotations(annotations map[string]string) *Builder {
```

Comment:
if I'm not wrong, you want to have single function that does override for annotations and one for labels etc.
In that case, we can have all with New like WithAnnotationsNew, WithLabelsNew etc

Reply:
Done